### PR TITLE
OSOE-603: ⭐ Upgrade to pnpm@8

### PR DESCRIPTION
**Pull request by sarahelsaig**
[OSOE-603](https://lombiq.atlassian.net/browse/OSOE-603)

[OSOE-603]: https://lombiq.atlassian.net/browse/OSOE-603?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -20,6 +20,7 @@ jobs:
       ui-test-parallelism: 0
       build-create-binary-log: "true"
       blame-hang-timeout: "5m"
+      build-verbosity: normal
```
Comment (Piedone) [inline .github/workflows/build-and-test.yml]: Isn't the default suitable?

File: package.json [hunk 2]
```diff
@@ -1,11 +1,11 @@
 {
   "private": true,
   "devDependencies": {
-    "eslint": "^8.25.0",
+    "eslint": "^8.45.0",
```
Comment (Piedone) [inline package.json]: And update would be great, but all the other package.jsons are 8.25.0.

**Review by Piedone — CHANGES_REQUESTED** [review]

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (sarahelsaig) [reply .github/workflows/build-and-test.yml]: It was just diagnostics, I removed most but I guess forgot this one.

File: package.json (hunk 2 above)
Comment (sarahelsaig) [reply package.json]: It wasn't intentional, not sure why but sometimes it tries to update it despite the lock. Reverted.

**Review by Piedone — APPROVED** [review]

File: pnpm-lock.yaml
Comment (Piedone) [inline pnpm-lock.yaml:1]: A rebuild causes this for me:

image

So something apparently needs to be done with the package.json file.

**Review by Piedone — CHANGES_REQUESTED** [review]

File: pnpm-lock.yaml
Comment (sarahelsaig) [reply pnpm-lock.yaml:1]: That often happens for me, with this or that NE-using project. I usually just revert it.

Comment (Piedone) [reply pnpm-lock.yaml:1]: We shouldn't have to revert it. This, and only this file, happens on every rebuild for me.

Comment (sarahelsaig) [reply pnpm-lock.yaml:1]: Yeah, it do be like that. it's not part of this issue, but I will investigate soon.

**Comment by Piedone** [comment]
Thank you!

--
Zoltán Lehóczky
Co-founder and managing director
Lombiq Technologies Ltd.
https://lombiq.com
[image]<https://lombiq.com/>
Please note that we store your personal data in accordance with our privacy policy<https://lombiq.com/privacy-policy>, but you can anytime choose to withdraw your previous consent by writing to ***@***.***
________________________________

**Review by Piedone — APPROVED** [review]

**Review by Piedone — APPROVED** [review]

**Pull request merged**